### PR TITLE
Bug 1878434 - Add the `awesomebar.{engagement, abandonment}` events.

### PR DESCRIPTION
**Pull request by linabutler**
This commit:

* Sends the new `awesomebar.{engagement, abandonment}` events when the user finishes interacting with the awesomebar. These events match [iOS's events of the same name](https://github.com/mozilla-mobile/firefox-ios/pull/18452), and Desktop's `urlbar.{engagement, abandonment}` events.
* Removes the `engagement_abandoned` extra key for the `awesomebar.{sponsored, non_sponsored}_suggestion_impressed` events, since they're specific to AMP and Wikipedia suggestions, and are subsumed by the new events.

This is the first step toward unifying our awesomebar search telemetry on all our platforms.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1878434

### CODE REVIEW

File: fenix/app/metrics.yaml [hunk 1]
```diff
@@ -9243,10 +9241,49 @@ awesomebar:
...
+      - ttran@mozilla.com
+      - najiang@mozilla.com
+    expires: never
+    metadata:
+      tags:
+        - Search
+  abandonment:
+    type: event
+    description: |
+      The user dismisses the awesomebar without completing their search.
```
Comment (linabutler) [inline fenix/app/metrics.yaml:9270]:
```suggestion
      The user dismissed the awesomebar without completing their search.
```

**Comment by linabutler** [comment]
# Request for data collection review form

1) What questions will you answer with this data?

* What is the click-through rate for awesomebar searches?
* How often do users abandon their awesomebar searches?

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?

Product and business requirements, to align the definitions of engaged and abandoned awesomebar searches on all our platforms. We'll use this data for future search and Suggest experiments, including Enhanced Cross-Platform Suggest.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

Firefox Desktop records these events today, under [`urlbar.engagement`](https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/metrics/urlbar_engagement) and [`urlbar.abandonment`](https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/metrics/urlbar_abandonment), and there's an [already-approved PR](https://github.com/mozilla-mobile/firefox-ios/pull/18452) that adds them to Firefox for iOS.

Firefox for Android currently records abandonments only when a sponsored or non-sponsored suggestion is shown, while Desktop and iOS record discrete events for all abandoned search sessions.

4) Can current instrumentation answer these questions?

Partially; please see question (3).

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.

This PR changes the following measurements in the `events` ping:

| Measurement                                     | Description                                                                                                       | Data Collection Category | Tracking Bug # |
|-------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|--------------------------|----------------|
| `awesomebar.sponsored_suggestion_impressed`     | Removes the `engagement_abandoned` extra key.                                                                     | 2                        | 1878434        |
| `awesomebar.non_sponsored_suggestion_impressed` | Removes the `engagement_abandoned` extra key.                                                                     | 2                        | 1878434        |
| `awesomebar.engagement`                         | The user finished interacting with the awesomebar by tapping a search result, or entering a URL or a search term. | 2                        | 1878434        |
| `awesomebar.abandonment`                        | The user dismissed the awesomebar without completing their search.                                                | 2                        | 1878434        |

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.

This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).

7) How long will this data be collected?  Choose one of the following:

This data will be collected permanently. The Suggest team will be responsible for the permanent collections.

8) What populations will you measure?

All users.

9) If this data collection is default on, what is the opt-out mechanism for users?

These collections are Glean. The opt-out can be found in Settings.

10) Please provide a general description of how you will analyze this data.

General reporting on Looker and Redash, as well as experiment-specific analyses.

11) Where do you intend to share the results of your analysis?

The results will be shared internally with the Suggest team.

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection?

No.

**Comment by linabutler** [comment]
@dzeber This is the change we were talking about on Slack earlier.

**Review by MatthewTighe — APPROVED** [review]
Code LGTM once the data review is complete!

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
>     1. What questions will you answer with this data?
> 
> 
>     * What is the click-through rate for awesomebar searches?
> 
>     * How often do users abandon their awesomebar searches?
> 
> 
>     2. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
> 
> 
> Product and business requirements, to align the definitions of engaged and abandoned awesomebar searches on all our platforms. We'll use this data for future search and Suggest experiments, including Enhanced Cross-Platform Suggest.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
> Firefox Desktop records these events today, under [`urlbar.engagement`](https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/metrics/urlbar_engagement) and [`urlbar.abandonment`](https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/metrics/urlbar_abandonment), and there's an [already-approved PR](https://github.com/mozilla-mobile/firefox-ios/pull/18452) that adds them to Firefox for iOS.
> 
> Firefox for Android currently records abandonments only when a sponsored or non-sponsored suggestion is shown, while Desktop and iOS record discrete events for all abandoned search sessions.
> 
>     4. Can current instrumentation answer these questions?
> 
> 
> Partially; please see question (3).
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> This PR changes the following measurements in the `events` ping:
> Measurement 	Description 	Data Collection Category 	Tracking Bug #
> `awesomebar.sponsored_suggestion_impressed` 	Removes the `engagement_abandoned` extra key. 	2 	1878434
> `awesomebar.non_sponsored_suggestion_impressed` 	Removes the `engagement_abandoned` extra key. 	2 	1878434
> `awesomebar.engagement` 	The user finished interacting with the awesomebar by tapping a search result, or entering a URL or a search term. 	2 	1878434
> `awesomebar.abandonment` 	The user dismissed the awesomebar without completing their search. 	2 	1878434
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).
> 
>     7. How long will this data be collected?  Choose one of the following:
> 
> 
> This data will be collected permanently. The Suggest team will be responsible for the permanent collections.
> 
>     8. What populations will you measure?
> 
> 
> All users.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> These collections are Glean. The opt-out can be found in Settings.
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
> General reporting on Looker and Redash, as well as experiment-specific analyses.
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
> The results will be shared internally with the Suggest team.
> 
>     12. Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection?
> 
> 
> No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

lina@mozilla.com, ttran@mozilla.com and najiang@mozilla.comwill be responsible for monitoring the data over time

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Pull request merged**